src/virtual_elements.ts

```diff
@@ -40,6 +40,49 @@ const ATTRIBUTES_OFFSET = 3;
 const prevAttrsMap = createMap();
 
 
+/**
+ * Removes an attribute name/value pair.
+ * @param attrsArr The array of attributes name/value pairs.
+ * @param name The name of the attribute to remove.
+ * @return The previous value, if any.
+ */
+// tslint:disable-next-line:no-any
+function removeAttr(attrsArr: any[], name: string): boolean {
+  for (let i = attrsArr.length - 2; i >= 0; i -= 2) {
```

Comment:
Why reverse iteration?

Reply:
It is common to reverse iterate when removing items from an array so that you don't need to worry about not incrementing the index when removing items.

Reply:
Note that if they are perfectly reversed (like I think they will be), the current code is an O(n<sup>3</sup>) alg.